service/resource/legacyv2/adapter/vpc.go

```diff
@@ -30,46 +30,24 @@ func VPCID(clients Clients, name string) (string, error) {
 	return *output.Vpcs[0].VpcId, nil
 }
 
-func SubnetID(clients Clients, name string) (string, error) {
-	describeSubnetInput := &ec2.DescribeSubnetsInput{
-		Filters: []*ec2.Filter{
-			{
-				Name: aws.String(fmt.Sprintf("tag:%s", tagKeyName)),
-				Values: []*string{
-					aws.String(name),
-				},
-			},
-		},
-	}
-	output, err := clients.EC2.DescribeSubnets(describeSubnetInput)
-	if err != nil {
-		return "", microerror.Mask(err)
-	}
-	if len(output.Subnets) > 1 {
-		return "", microerror.Mask(tooManyResultsError)
-	}
-
-	return *output.Subnets[0].SubnetId, nil
-}
-
-func RouteTableID(clients Clients, name string) (string, error) {
-	input := &ec2.DescribeRouteTablesInput{
+func VpcCIDR(clients Clients, vpcID string) (string, error) {
```

Comment:
This function gets the CIDR of the host cluster for the sec group rules.

Reply:
I think `clients` here are currently configured to work with the guest cluster, see how they are injected from `awsClients` https://github.com/giantswarm/aws-operator/blob/master/service/framework.go#L229, how `awsClients` is defined using `awsConfig` https://github.com/giantswarm/aws-operator/blob/master/service/framework.go#L173 and where `awsConfig` comes from https://github.com/giantswarm/aws-operator/blob/master/service/framework.go#L146. 

IMO we could add a `HostClients` member to the resource (only an EC2 client would do), WDYT? This will be needed for porting the VPC and VPCPeering resources too.

Reply:
Good catch on this. Yes I think adding a HostClients member is good especially as we only need the EC2 client. I'll implement that. Thx!